src/Sylius/Bundle/TranslationBundle/Doctrine/ODM/MongoDB/TranslatableResourceRepository.php

```diff
@@ -80,19 +78,43 @@ protected function applyCriteria(QueryBuilder $queryBuilder, array $criteria = n
         }
 
         foreach ($criteria as $property => $value) {
-            if (in_array($property, $this->translatableFields)) {
-                $property = 'translations.'.$this->localeProvider->getCurrentLocale().'.'.$property;
-            }
-
             if (is_array($value)) {
                 $queryBuilder
-                    ->field($property)->in($value)
+                    ->field($this->getPropertyName($property))->in($value)
                 ;
             } elseif ('' !== $value) {
                 $queryBuilder
-                    ->field($property)->equals($value)
+                    ->field($this->getPropertyName($property))->equals($value)
                 ;
             }
         }
     }
+
+    /**
+     * {inheritdoc}.
+     */
+    protected function applySorting(QueryBuilder $queryBuilder, array $sorting = null)
```

Comment:
Why not simply initialize ```$sorting``` as an empty array?

```php
protected function applySorting(QueryBuilder $queryBuilder, array $sorting = array())
```

We could now simplify method by deleting check for null variable.